### DISCUSSION TITLE
Use v3 with Tide?

### DISCUSSION THREAD

**Discussion by jgreenbaum** [http-rs/async-session#50] [Announcements]
It seems that the removal of async-std in v3 has broken the use of async-session with tide. If you try a simple tide program like the one in the async-mongodb-session/examples but with async-sqlx-session v0.4.0 you get build errors like

```the trait `SessionStore` is not implemented for `SqliteSessionStore` ```

Moving async-sqlx-session back to async-session v2.0.1 from v3.0.0 makes the build error go away. I think this is a bug in async-session not async-sqlx-session because moving async-mongodb-session to async-session v3.0.0 from v2.0.1 results in the same build errors but for MongodbSessionStore, and I just don't see anything else in the latest async-sqlx-session that seems like it could be the cause.

Thoughts? Anything more I can do to help troubleshoot?

**Comment by jbr** [comment]
This is a known issue and will be fixed in the next tide release

**Comment by jgreenbaum** [comment]
Thanks, I thought it might be a roadmap thing when I saw the commits putting things behind an async-std feature. What is a better forum than here next time I see something like this that might just be WIP?

**Comment by jbr** [comment]
This is a totally reasonable place to file an issue. When I bumped async-session, I expected tide would be having more frequent releases, but there hasn't been a tide release in quite a while. Hopefully there will be one soon, and there is reason to expect that this may be the case.

**Comment by xyzshantaram** [comment]
> Moving async-sqlx-session back to async-session v2.0.1 from v3.0.0 makes the build error go away

@jgreenbaum could you elaborate on how exactly to do this? I'm quite new to Tide and Rust in general and I've just run into this issue.

~~Edit: Figured it out, just had to manually specify the crate and its version in Cargo.toml.~~

~~Edit: Specifying the crate didn't work...~~

Edit: Finally actually figured it out - I just cloned async-sqlx-session, fixed the version of async-session, and used `path=` in Cargo.toml to use that version instead of the one from crates.io.

Thanks for the tip to downgrade async-session.

**Comment by jgreenbaum** [comment]
The workaround I use is to specify version 0.3.0 explicitly in my Cargo.toml:
 
```async-sqlx-session = { version = "0.3.0", features = ["pg", "sqlite"] }```

**Comment by jbr** [comment]
Converting this issue to a discussion, as it is relevant to this crate but there is nothing further we can do here (tide needs a new release but is currently minimally maintained)
